### DISCUSSION TITLE
Feature Request: Theming

### DISCUSSION THREAD

**Discussion by ansidev** [markdoc/markdoc#282] [Q&A] [answered]
Currently, many similar document-generation tools support this feature. Ex: NextJS, Astro, VuePress, VitePress,...

A theming system separates the theme from the content, so we can reuse a theme for many sites easily.

As a developer, I want to customize the current theme of the document site.

**Features**
- Standard theme folder structure.
	- Maintainers/Contributors can maintain a theme easily since it follows standards.
- Easy to customize.
- Easy to integrate with popular Framework, ex. TailwindCSS, Bootstrap.
- Easy to test.
- A skeleton theme + CLI to init a skeleton theme quickly. It can save time to develop a new theme.
- If these features were supported via the official libraries (utility functions), it's perfect since they are popular among documentation sites:
	- Darkmode.
	- Table of contents.
	- Others (TBD).
- Good documentation.
- A site that lists official and community themes. It saves time in for searching a theme.
- Others (TBD).

**Comment by segphault** [comment]
Can you elaborate on the specific functionality you would want for a Markdoc theming system? You can use tag and node schemas to completely control the markup that the Markdoc renderer outputs, so it already provides extensive control over presentation.

**Comment by ansidev** [comment]
@segphault I've updated the issue description. FYI!

**Comment by mfix-stripe** [comment, marked as answer]
@ansidev Markdoc is completely agnostic to how you style your site. You can use Markdoc to generate React or HTML, and use any of the systems above to handle theming.

For example, you could hook up a Next.js site with our Markdoc plugin ([`@markdoc/next.js`](https://www.npmjs.com/package/@markdoc/next.js)), and then add in Tailwind support to your entire Next.js site to style the content: https://github.com/vercel/next.js/tree/canary/examples/with-tailwindcss

In short: you should handle your theming using any React or HTML based theming solution, and it will work with Markdoc.
